perun-core/src/test/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntryIntegrationTest.java

```diff
@@ -825,6 +826,35 @@ public void getHostsWhenFacilityNotExists()throws Exception{
 
 	}
 
+	@Test
+	public void getEnrichedHosts() throws Exception {
+		System.out.println(CLASS_NAME + "getEnrichedHosts");
+
+		createdHost = facilitiesManagerEntry.addHosts(sess, hosts, facility).get(0);
+		// set this host for deletion - host is created after adding to facility !!
+		hostsForDeletion.add(hosts.get(0));
+
+		Attribute attr = new Attribute();
+		attr.setNamespace("urn:perun:host:attribute-def:opt");
+		attr.setFriendlyName("host-test-for-list-of-names-attribute");
+		attr.setType(String.class.getName());
+		attr.setValue("HostAttributeForList");
+		perun.getAttributesManagerBl().createAttribute(sess, attr);
+		perun.getAttributesManagerBl().setAttribute(sess, createdHost, attr);
+
+		List<String> attrNames = new ArrayList<>();
+		attrNames.add(attr.getName());
+
+		List<Attribute> hostAttributes = new ArrayList<>();
+		hostAttributes.add(attr);
+		EnrichedHost actualEnrichedHost = new EnrichedHost(createdHost, hostAttributes);
+
+		List<EnrichedHost> expectedEnrichedHosts = facilitiesManagerEntry.getEnrichedHosts(sess, facility, attrNames);
+		EnrichedHost expectedHost = expectedEnrichedHosts.get(0);
+
+		assertEquals("Created and returned enrichedHost should be the same", expectedHost, actualEnrichedHost);
```

Comment:
Yo changed the logic. So now you need to check if there are expected attributes manually.

Reply:
Yes, it should be fixed now as well.